### PR TITLE
Remove injections & remove MockedClusterServiceModule

### PR DESCRIPTION
**Pull request by mfussenegger**

### CODE REVIEW

**Comment by mfussenegger** [comment]
@matriv please review

File: sql/src/main/java/io/crate/planner/statement/DeleteStatementPlanner.java [hunk 1]
```diff
@@ -50,14 +50,12 @@
 import io.crate.planner.projection.Projection;
 import io.crate.types.DataTypes;
 import org.elasticsearch.cluster.routing.Preference;
-import org.elasticsearch.common.inject.Singleton;
 
 import java.util.*;
 
-@Singleton
 public class DeleteStatementPlanner {
```
Comment (matriv) [inline sql/src/main/java/io/crate/planner/statement/DeleteStatementPlanner.java]: Maybe also final?

File: sql/src/test/java/io/crate/analyze/where/WhereClauseAnalyzerTest.java [hunk 2]
```diff
@@ -23,86 +23,60 @@
...
+    private final Routing twoNodeRouting = new Routing(TreeMapBuilder.<String, Map<String, List<Integer>>>newMapBuilder()
+        .put("nodeOne", TreeMapBuilder.<String, List<Integer>>newMapBuilder().put("t1", Arrays.asList(1, 2)).map())
+        .put("nodeTow", TreeMapBuilder.<String, List<Integer>>newMapBuilder().put("t1", Arrays.asList(3, 4)).map())
+        .map());
     private final TransactionContext transactionContext = new TransactionContext();
-
-    @Mock
-    private SchemaInfo schemaInfo;
-    private Functions functions;
+    private SQLExecutor e;
```
Comment (matriv) [inline sql/src/test/java/io/crate/analyze/where/WhereClauseAnalyzerTest.java]: would rename it to sqlExecutor or executor

File: sql/src/test/java/io/crate/planner/consumer/NestedLoopConsumerTest.java [hunk 3]
```diff
@@ -80,79 +71,51 @@
...
+        .build();
 
     @Rule
     public ExpectedException expectedException = ExpectedException.none();
 
-    private final ClusterService clusterService = mock(ClusterService.class);
     private NestedLoopConsumer consumer;
     private Planner.Context plannerContext;
-    private TableStatsService statsService;
+    private SQLExecutor e;
```
Comment (matriv) [inline sql/src/test/java/io/crate/planner/consumer/NestedLoopConsumerTest.java]: same here

File: sql/src/test/java/io/crate/testing/SQLExecutor.java [hunk 4]
```diff
@@ -150,14 +165,24 @@ public Builder addBlobTable(BlobTableInfo tableInfo) {
...
+        this.functions = functions;
         this.analyzer = analyzer;
+        this.planner = planner;
+    }
+
+    public Functions functions() {
+        return functions;
     }
 
     private <T> T analyze(String stmt, ParameterContext parameterContext) {
```
Comment (matriv) [inline sql/src/test/java/io/crate/testing/SQLExecutor.java]: maybe use ? extends AnalyzedStatement here and for the other analyze()

File: sql/src/test/java/io/crate/testing/SQLExecutor.java [hunk 5]
```diff
@@ -182,4 +207,11 @@ private SQLExecutor(Analyzer analyzer) {
     public <T> T analyze(String statement, Object[][] bulkArgs) {
         return analyze(statement, new ParameterContext(Row.EMPTY, Rows.of(bulkArgs)));
     }
+
+    public <T> T plan(String statement, UUID jobId, int softLimit, int fetchSize) {
```
Comment (matriv) [inline sql/src/test/java/io/crate/testing/SQLExecutor.java]: also here <? extends Plan> ?

File: sql/src/test/java/io/crate/planner/consumer/ManyTableConsumerTest.java [hunk 6]
```diff
@@ -27,58 +27,31 @@
...
-            .add(new PredicateModule())
-            .add(new RepositorySettingsModule())
-            .add(new OperatorModule())
-            .createInjector();
-        analyzer = injector.getInstance(Analyzer.class);
-    }
+
+public class ManyTableConsumerTest extends CrateUnitTest {
+
+    private SQLExecutor e = SQLExecutor.builder(new NoopClusterService()).enableDefaultTables().build();
```
Comment (matriv) [inline sql/src/test/java/io/crate/planner/consumer/ManyTableConsumerTest.java]: same here

File: sql/src/test/java/io/crate/analyze/where/WhereClauseAnalyzerTest.java (hunk 2 above)
Comment (matriv) [inline sql/src/test/java/io/crate/analyze/where/WhereClauseAnalyzerTest.java]: Ok, then consistency is more important I just don't like short names like that for class attributes, for local variables it's ok and e is usually what we use for exception that's why I mentioned, but I leave up to you!

**Pull request merged**